### ISSUE TITLE
Incorrect parsing in `read_tables._extract_fixed_width_field_data()`

### ISSUE THREAD

**Issue by eischaefer** [Small-Bodies-Node/pds4_tools#92]
The function `read_tables._extract_fixed_width_field_data()` is constructed to prefer a few optimized special cases and fall back to general-case code. However, that general code does not correctly extract from `table_byte_data` in my case. Specifically, in my case, the entire table is a single group with a single, repeated field. Since the entirety of `table_byte_data` should be appended to `extracted_data`, I can perform a partial verification of correct parsing by inserting code into the general-case section that creates a zeroed array and then increments each value position as it is used:
```python
a = np.zeros(len(table_byte_data), np.int64)  # New
...
    extracted_data.append(table_byte_data[start_byte:stop_byte])  # Original
    a[start_byte:stop_byte] += 1  # New
```

After extraction, `np.all(a == 1)` should be `True`, indicating that each byte was read exactly once, but instead:
```python
a.min() == 0
a.max() == 8
(a == 1).sum() == 256  # Values read exactly once.
a.size == 384_000_000  # Count of all values.
(a.mean() < 1) == True  # Because some slices extend beyond end.
```

Sufficient for my (very special) use case, I added this code at the start of  `read_tables._extract_fixed_width_field_data()`:
```python
    # Most simplified, sped up, case for fields in a single group that 
    # spans entire table
    if ((field_location == 0) and
        (len(repetition_lengths) == 1) and 
        (field_length*repetition_lengths[0] == record_length)):

        # Loop over each field in each record
        for start_byte in range(0, len(table_byte_data), field_length):
            stop_byte = start_byte + field_length
            
            extracted_data.append(table_byte_data[start_byte:stop_byte])
     
        return
```

Analysis with array `a`, as further above, results in  `np.all(a == 1) == True`.

That case may be too special to directly support in the production code, but I provide it for reference. Certainly, the general-case code still needs debugging.

**Comment by LevN0** [comment]
Ethan, thank you for reporting.

For ease of debugging, could you include your label and data file? Either upload it here or email me with it by clicking on the link with my name [here](https://pdssbn.astro.umd.edu/tools/tools_readPDS.shtml).

**Comment by eischaefer** [comment]
> Ethan, thank you for reporting.
> 
> For ease of debugging, could you include your label and data file? Either upload it here or email me with it by clicking on the link with my name [here](https://pdssbn.astro.umd.edu/tools/tools_readPDS.shtml).

I have now e-mailed you links to the label and data.

**Comment by LevN0** [comment]
Thank you. We should confirm that we agree in terms of how group fields should be defined according to the PDS4 standard. 

Could you take a look at how the [record_length](https://sbnwiki.astro.umd.edu/wiki/Filling_Out_the_Table_Binary_Data_Structure#.3Crecord_length.3E), the [group_location](https://sbnwiki.astro.umd.edu/wiki/Filling_Out_the_Table_Binary_Data_Structure#.3Cgroup_location.3E) and [group_length](https://sbnwiki.astro.umd.edu/wiki/Filling_Out_the_Table_Binary_Data_Structure#.3Cgroup_length.3E) fields are defined in the aforementioned links, and let me know if this is also how your values are expressed in the label?

**Comment by eischaefer** [comment]
Thank you for your attention. There is no bug.

I am migrating some PDS3 labels to PDS4 and the original label incorrectly provides ITEM_BYTES and ITEM_OFFSET in bits, not bytes. The software that I created to facilitate that migration correctly catches the misstatement for ITEM_BYTES, and consequently <field_length> is correct, but does not correct ITEM_OFFSET, and therefore <group_length> was incorrect in the initial PDS4 label.

I have since fixed that bug and the updated PDS4 label is correctly read by the PDS4 Tools code.

Incidentally, I find your code very clean and well documented! Thank you for that.
